src/pages/Admin/Admin_v2.tsx

```diff
@@ -0,0 +1,454 @@
+import AuthenticationModal from "components/organisms/AuthenticationModal";
+import WithNavigationBar from "components/organisms/WithNavigationBar";
+import dayjs from "dayjs";
+import advancedFormat from "dayjs/plugin/advancedFormat";
+import "firebase/storage";
+import { useKeyedSelector, useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+import React, { useState, useEffect, useRef, useCallback } from "react";
+import { useFirestoreConnect } from "react-redux-firebase";
+import {
+  Link,
+  Route,
+  Switch,
+  useLocation,
+  useParams,
+  useRouteMatch,
+  useHistory,
+} from "react-router-dom";
+import { AdminVenueDetailsPartProps, VenueEvent } from "types/VenueEvent";
+import { WithId } from "utils/id";
+import {
+  canHaveSubvenues,
+  canBeDeleted,
+  canHaveEvents,
+  canHavePlacement,
+} from "utils/venue";
+import "./Admin.scss";
+import AdminEventModal from "./AdminEventModal";
+import { venueInsideUrl } from "utils/url";
+import { AdminVenuePreview } from "./AdminVenuePreview";
+import { isCampVenue } from "types/CampVenue";
+import { useQuery } from "hooks/useQuery";
+import { VenueTemplate } from "types/VenueTemplate";
+import VenueDeleteModal from "./Venue/VenueDeleteModal";
+import { PlayaContainer } from "pages/Account/Venue/VenueMapEdition";
+import {
+  PLACEABLE_VENUE_TEMPLATES,
+  PLAYA_IMAGE,
+  PLAYA_VENUE_SIZE,
+  PLAYA_VENUE_STYLES,
+  PLAYA_VENUE_NAME,
+  PLAYA_WIDTH,
+  PLAYA_HEIGHT,
+} from "settings";
+import AdminEditComponent from "./AdminEditComponent_v2";
+import { VenueOwnersModal } from "./VenueOwnersModal";
+import useRoles from "hooks/useRoles";
+import { IS_BURN } from "secrets";
+import EventsComponent from "./EventsComponent";
+import AdminDeleteEvent from "./AdminDeleteEvent";
+
+dayjs.extend(advancedFormat);
+
+type VenueListProps = {
+  selectedVenueId?: string;
+  roomIndex?: number;
+};
+
+const VenueList: React.FC<VenueListProps> = ({
+  selectedVenueId,
+  roomIndex,
+}) => {
+  const venues = useSelector((state) => state.firestore.ordered.venues);
+
+  if (!venues) return <>Loading...</>;
+
+  return (
+    <>
+      <div className="page-container-adminsidebar-title title">My Venues</div>
+      <div className="page-container-adminsidebar-top">
+        <Link to="/admin/venue/creation" className="btn btn-primary">
+          Create a venue
+        </Link>
+      </div>
+      <ul className="page-container-adminsidebar-venueslist">
+        {venues.map((venue, index) => (
+          <li
+            key={index}
+            className={`${selectedVenueId === venue.id ? "selected" : ""} ${
+              canHaveSubvenues(venue) ? "camp" : ""
+            }`}
+          >
+            <Link to={`/admin/venue/${venue.id}`}>{venue.name}</Link>
+            {isCampVenue(venue) && (
+              <ul className="page-container-adminsidebar-subvenueslist">
+                {venue.rooms.map((room, idx) => (
```

Comment:
I think the typings for the rooms are messed up and this needs to be checked with `venue?.rooms`